### ISSUE TITLE
tab rows / columns

### ISSUE THREAD

**Issue by PimpUigi** [Noitidart/Homepage-New-Tab#3]
In the Fx31 beta, it kept auto detecting how many rows of tabs would be on the new tab page. Now on release, it's not doing that and always shows 9.
I know your add on has an option to force it to a certain amount, but I was wondering if you would be able to make it auto select according to resolution like it's doing on beta / aurora / nightly releases.
Not sure what code does this, I wasn't able to figure it out with dom inspector.

**Comment by Noitidart** [comment]
Hewy man thanks for the report. I don't understand. Can you please post screenshot.

**Comment by PimpUigi** [comment]
I know, it's hard to explain. Basically no matter how you resize the window, the Fx31 beta was automatically adding / removing tabs instead of adding a scroll bar.

Also they were always sized the same way, instead of the tiles getting wider / taller depending. I am not sure why those things aren't in release.

**Comment by Noitidart** [comment]
Ah I got you now. Ok that's weird I'll look into this. Also can you please test enabling the search field, I think it's margin is too much in the release version for some reason.

**Comment by PimpUigi** [comment]
Search field?

**Comment by Noitidart** [comment]
Hm weird you dont see one? Maybe its disabled by default I forget, can you please go to prefrerences of the add-on and enable the field.

Date: Tue, 22 Jul 2014 20:16:12 -0700

**Comment by PimpUigi** [comment]
I attached a picture to the first email, was it there?

When I right click a tab, and go to "This Frame" and tell it to reload frame, I have a bug where an extra one appears. (in this screenshot)

**Comment by Noitidart** [comment]
No man screenshots dont show over email for some reason :(

**Comment by PimpUigi** [comment]
ohhhh….

http://s12.postimg.org/eop3u2efv/newtab.png

This is the main thing I was talking about. Even though I have 9x9 set in your options, the current beta / aurora / nightly channels all make it neat and laid out nicely.

http://s27.postimg.org/ptjpojmf5/Search_Field.png

This is the glitch I was talking about when I try to reload frame.

**Comment by Noitidart** [comment]
Ah yuck. So in this imgae here the fix is to make sure this field stays hidden. This happens only after reload?
![](http://s27.postimg.org/ptjpojmf5/Search_Field.png)

---

And the problem in this image here It should show proper number of panels.

![](http://s12.postimg.org/eop3u2efv/newtab.png)

---

**Comment by PimpUigi** [comment]
No problem with the number of panels, in fact it's better.

Beta / Nightly / Aurora's current system with the tiles is to not let the scroll bar appear no matter what. 

This makes the tiles more intuitive. But for some reason, once it hit release…the tiles went back to always showing 9 tiles and a scroll bar.

Also Nightly keeps the tiles the same size no matter what. All good things. : )

**Comment by Noitidart** [comment]
Yeah man interesting. I only have beta installed and it was beutiful in that it resized as I resized window until it got too small than it would keep it at a readable level and then cut off the others.

I'll install release version and test. Thanks man, I never though that if it worked in beta that it might not work in release. Very interesting.

**Comment by PimpUigi** [comment]
Could be a bug, maybe they'll fix it like a Fx31.1 release like they sometimes do.

**Comment by Noitidart** [comment]
Haha hopefully that would save me work :P

**Comment by PimpUigi** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=994915

Thought you might like to know that one of the requested features is already planned. Could save you some work.

**Comment by Noitidart** [comment]
Thanks so much man!
